shell/models/management.cattle.io.project.js

```diff
@@ -2,24 +2,38 @@ import { DEFAULT_PROJECT, SYSTEM_PROJECT } from '@shell/config/labels-annotation
 import { MANAGEMENT, NAMESPACE, NORMAN } from '@shell/config/types';
 import HybridModel from '@shell/plugins/steve/hybrid-class';
 
-function clearResourceQuotas(val, type) {
-  if (val[type]?.limit && Object.keys(val[type].limit).length) {
-    Object.keys(val[type].limit).forEach((key) => {
-      if (!val[type].limit[key]) {
-        delete val[type].limit[key];
-      }
-    });
-  }
+function clearUnusedResourceQuotas(spec, types) {
+  types.forEach((type) => {
+    if (spec[type]?.limit && Object.keys(spec[type].limit).length) {
+      Object.keys(spec[type].limit).forEach((key) => {
+        if (!spec[type].limit[key]) {
+          delete spec[type].limit[key];
+        }
+      });
+    }
 
-  if ( val[type]?.limit && !Object.keys(val[type].limit).length ) {
-    delete val[type].limit;
-  }
+    if (spec[type]?.usedLimit && Object.keys(spec[type].usedLimit).length) {
+      Object.keys(spec[type].usedLimit).forEach((key) => {
+        if (!spec[type].usedLimit[key]) {
+          delete spec[type].usedLimit[key];
+        }
+      });
+    }
 
-  if ( val[type] && !Object.keys(val[type]).length ) {
-    delete val[type];
-  }
+    if ( spec[type]?.limit && !Object.keys(spec[type].limit).length ) {
+      spec[type].limit = null;
+    }
 
-  return val[type];
+    if ( spec[type]?.usedLimit && !Object.keys(spec[type].usedLimit).length ) {
+      spec[type].usedLimit = null;
+    }
+
+    if ( spec[type] && Object.keys(spec[type]).length && Object.keys(spec[type]).every( k => spec[type][k] === null ) ) {
+      spec[type] = null;
+    }
+  });
```

Comment:
This is more of a style preference, but there is a utility method [`isEmpty`](https://github.com/rancher/dashboard/blob/master/shell/utils/object.js#L98-L104) that may help clean this up a bit.

Reply:
Changes done. Thanks for the feedback!